webscrapping101/scrape.ipynb

```diff
@@ -0,0 +1,249 @@
+      "{'CourseCode': 'EECS 183', 'CourseName': 'Elementary Programming Concepts', 'Credit': '4', 'Link': 'https://atlas.ai.umich.edu/course/EECS%20183/', 'Note': 'Prerequisite: None. (Credit for only one: EECS 180, EECS 183, ENGR 101 or ENGR 151) (4 credits)'}\n",
+      "{'CourseCode': 'EECS 198', 'CourseName': 'Special Topics', 'Credit': '', 'Link': 'https://atlas.ai.umich.edu/course/EECS%20198/', 'Note': 'Advisory Prerequisite: Permission of instructor. (1-4 credits)'}\n",
+      "{'CourseCode': 'EECS 200', 'CourseName': 'Electrical Engineering Systems Design I', 'Credit': '', 'Link': 'https://atlas.ai.umich.edu/course/EECS%20200/', 'Note': 'Advisory Prerequisite: ENGR 100 or ENGR 101 or ENGR 151 or EECS 180 or EECS 280. Preceded or accompanied by: EECS 215 Minimum grade requirement of “C” for advised prerequisites. (2'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "import re\n",
+    "data = []\n",
+    "for c, p, l in zip(course, prereq, link):\n",
+    "    data_entry = {}\n",
+    "    course_split = c.split(\".\", 1) # Split at first occurance of a period\n",
+    "    if len(course_split) != 2: continue # Skip instances where there is bad course format. \n",
+    "    data_entry[\"CourseCode\"] = course_split[0]\n",
+    "    data_entry[\"CourseName\"]= course_split[1].lstrip(\" \") # Remove leading space.\n",
+    "\n",
+    "    credit = re.findall(r'\\((\\d+)\\s+credits\\)', p)\n",
+    "    \"\"\"\n",
+    "    the regular expression \\((\\d+)\\s+credits\\) matches any sequence of digits (\\d+)\n",
+    "    that are directly after a parenthesis and followed by one or more spaces (\\s+)\n",
+    "    and the word \"credits\". The parentheses around \\d+ create a group that \n",
+    "    findall() returns as a list.\n",
+    "    \"\"\"\n",
+    "    data_entry[\"Credit\"] = \" \".join(credit)\n",
+    "    data_entry[\"Link\"] = l\n",
+    "    data_entry[\"Note\"] = p\n",
+    "    data.append(data_entry)\n",
+    "\n",
+    "for i in range(5):\n",
+    "    print(data[i])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, lets put our data into a CSV file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 88,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import csv\n",
+    "\n",
+    "fields = [\"CourseCode\", \"CourseName\", \"Credit\", \"Link\", \"Note\"]\n",
+    "\n",
+    "# Open (or create) a CSV file with write permissions ('w')\n",
+    "with open('eecs_course.csv', 'w', newline='\\n') as csvfile:\n",
+    "    writer = csv.DictWriter(csvfile, fieldnames=fields)\n",
+    "    writer.writeheader()\n",
+    "    for entry in data:\n",
+    "        writer.writerow(entry)"
```

Comment:
I think converting to a dataframe and then just `write_csv` is cleaner. Up to you

Reply:
Yes, much cleaner